src/calculate_bucket.hpp

```diff
@@ -306,12 +309,14 @@ class FxCalculator {
             for (uint8_t i = 0; i < kExtraBitsPow; i++) {
                 uint16_t r_target = L_targets[parity][r][i];
                 for (size_t j = 0; j < rmap[r_target].count; j++) {
-                    matches.push_back(std::make_pair(pos_L, rmap[r_target].pos + j));
+                    if(idx_L != NULL) {
```

Comment:
can `idx_L` really be null here? if it can, what's the expected behavior, to *just* increment the counter? This check probably won't be expensive here since it will make the same branch every time through the loop. But it looks a bit suspicious to support it being null to begin with. It could be tested once and have two separate loops for the two behaviors, to hoist the check out of the inner loop

Reply:
Yes, in tests we only need the count